src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,338 @@
+		<template x-if="mapping.targetType == 'service'">
+			<div class="mb-6 space-y-6">
+				@componentForm.SelectInput("targetValue", "Service Name", "mapping.targetValue", getOnlyServiceNames(), true, false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'url'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "targetValue", "Destination URL", "mapping.targetValue", false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'inline-html'">
+			<div class="mb-6 space-y-6">
+				@componentForm.TextArea("targetValue", "Inline HTML", "mapping.targetValue", false)
+				@componentMappings.ResponseCodeSelectInput("targetHttpResponseCode", "Response Code", "mapping.targetHttpResponseCode")
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'response-code'">
+			<div class="mb-6 space-y-6">
+				@componentMappings.ResponseCodeSelectInput("targetValue", "Response Code", "mapping.targetValue")
+				<input type="hidden" x-model="mapping.targetValue" x-bind:value="mapping.targetHttpResponseCode"/>
+			</div>
+		</template>
+		<section class="mb-6 w-full rounded-md">
+			<details class="bg-os-500 group rounded-md p-4 group-open:bg-opacity-30" x-bin:open="isAdvancedSettingsClosed">
+				<summary class="relative flex cursor-pointer list-none items-center pr-8 text-neutral-50 text-opacity-80 focus-visible:outline-none group-open:text-opacity-100 group-hover:text-opacity-100 [&::-webkit-details-marker]:hidden">
+					<i class="ph-duotone ph-gear mr-2 text-2xl"></i>
+					<span class="font-bold">Advanced Settings</span>
+					<i class="ph-bold ph-plus absolute right-0 top-1.5 text-xs transition duration-300 group-open:rotate-45"></i>
+				</summary>
+				<div class="mt-5 py-2">
+					@componentMappings.MatchPatternSelectInput("matchPattern", "Match Pattern", "mapping.matchPattern")
+				</div>
+			</details>
+		</section>
+		<template x-if="virtualHost.hostname == '' || ((mapping.targetType != 'response-code' && mapping.targetType != 'static-files') && mapping.targetValue == '')">
+			@componentForm.SubmitButton("Create", "ph-check-fat", "", false, true)
+		</template>
+		<template x-if="virtualHost.hostname != '' && ((mapping.targetType == 'response-code' || mapping.targetType == 'static-files') || mapping.targetValue != '')">
+			@componentForm.SubmitButton("Create", "ph-check-fat", "closeCreateMappingModal()", false, false)
+		</template>
+	</form>
+}
+
+templ CreateMappingModal(vhostsWithMappings []dto.VirtualHostWithMappings) {
+	@componentStructural.Modal(
+		"Create Mapping", "isCreateMappingModalOpen", "closeCreateMappingModal()",
+		"",
+	) {
+		@MappingRegularForm(vhostsWithMappings)
+	}
+}
+
+templ CreateMappingFromVhostModal(vhostsWithMappings []dto.VirtualHostWithMappings) {
+	@componentStructural.Modal(
+		"Create Mapping", "isCreateMappingFromVhostModalOpen",
+		"closeCreateMappingModal()", "virtualHost.hostname",
+	) {
+		@MappingRegularForm(vhostsWithMappings)
+	}
+}
```

Comment:
Why are you passing the entire vhostsWithMappings to these modals?

Reply:
Useless. Removed!